src/main/java/net/mcreator/ui/component/entries/JSimpleEntriesList.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2023, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.component.entries;
+
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.help.IHelpContext;
+
+import javax.swing.*;
+import java.util.List;
+import java.util.Objects;
+
+public abstract class JSimpleEntriesList<T extends JSimpleListEntry<U>, U> extends JSingleEntriesList<T, U> {
+
+	public JSimpleEntriesList(MCreator mcreator, IHelpContext gui) {
+		super(mcreator, gui);
+
+		add.addActionListener(e -> {
+			@SuppressWarnings("unchecked") T entry = (T) newEntry(entries, entryList);
+			entry.reloadDataLists();
+			entryList.add(entry);
+			entry.setEnabled(this.isEnabled());
+			registerEntryUI(entry);
+			entryAddedByUserHandler();
+		});
+	}
+
+	public void entryAddedByUserHandler() {
+	}
+
+	public abstract JSimpleListEntry<U> newEntry(JPanel parent, List<T> entryList);

```

Comment:
This can actually return a `T` value, can't it? (Current way also allows cases when CCE gets thrown)

Reply:
Generics in Java are not covariant (https://www.freecodecamp.org/news/understanding-java-generic-types-covariance-and-contravariance-88f4c19763d2/):

![image](https://github.com/MCreator/MCreator/assets/16374228/b1f5410a-b122-437c-9478-379a6ac85866)

Reply:
Oh, yeah, given `T` is not subtype of `JSimpleListEntry<? extends U>` but `JSimpleListEntry<U>`, one can not change return value of this method to be of type `T` already. Forgot that, sorry :man_shrugging: